zipkin-thrift/src/main/thrift/com/twitter/zipkin/zipkinCore.thrift

```diff
@@ -204,5 +204,19 @@ struct Span {
   6: list<Annotation> annotations, # all annotations/events that occured, sorted by timestamp
   8: list<BinaryAnnotation> binary_annotations # any binary annotations
   9: optional bool debug = 0       # if true, we DEMAND that this span passes all samplers
+  /**
+   * Microseconds from epoch, usually taken from the first annotation. Ex.
+   * SERVER_RECV.timestamp
+   */
+  10: optional i64 start_ts,
```

Comment:
PS I don't see this as complicating, particularly as I've touched all the
code in zipkin. The current codebase is a mess as it has implicit logic
around things like startTs and duration. This formalizes it, making it
possible to create an un-hairy implementation.